easier_net/easier_net.py

```diff
@@ -45,94 +40,68 @@ def __init__(
-            est.num_batches=self.num_batches
-            est.n_jobs=self.n_jobs
-
+        for est in range(self.n_estimators): #TODO: i -> est?
+            est=sier_net.SierNetEstimator(
+                num_classes=self.num_classes,
+                input_filter_layer=self.input_filter_layer,
+                n_hidden= self.n_hidden,
+                max_iters=self.max_iters,
+                max_prox_iters=self.max_prox_iters,
+                full_tree_pen=self.full_tree_pen,
+                input_pen=self.input_pen,
+            ) 
+            
             estimators.append(est)
 
         return estimators
 
+
     def write_model(self, filename): #save and write model to file
-        #from fit_easier_net
-        meta_state_dict = self.estimators[0].get_params()
+        #save as .pt file
+        meta_state_dict = self.get_params()
         meta_state_dict["state_dicts"] = [
             est.net.state_dict() for est in self.estimators
         ]
         torch.save(meta_state_dict, filename)
-        self.estimators[0].net.get_net_struct()
-
-        #json way
-        # for est in self.estimators:
-        #     model_json = est.to_json() #model_fit_params_file is json
-        #     with open("model.json", "w") as json_file:
-        #         json_file.write(model_json)
-        #     print(f"[INFO]>>Save {model_json}.")
-
-        #pickle way
-        # #filename ex) "easiernet_estimators.pkl"
-        # with open(filename, 'wb') as file:
-        #     pickle.dump(self.estimators, file)
-
-    def load_all_models(self):
-        estimators=sier_net.SierNetEstimator()
-        #definitely wrong, need to load for each state_dicts
-        estimators.load_state_dict(torch.load(self.model_fit_params_file, map_location='cpu'), strict=False) #to avoid gpu ram surge
-
-
-        #loading json way
-        # estimators = []
-        # for i in range(self.n_estimators):
-        #     filename = 'models/model_' + str(i) + '.json'
-        #     est = model_from_json(self.model_fit_params_file)
-        #     estimators.append(est)
-        #     print(f"[INFO]>>loaded {filename}.")
-        # return estimators
-
-        #loading pickle way
-        with open(self.model_fit_params_file, 'rb') as file:
-            estimators = pickle.load(file) 
-            for est in self.estimators:
-                est.full_tree_pen = self.estimators["full_tree_pen"]
-                est.input_pen = self.estimators["input_pen"]
-                est.input_filter_layer = self.estimators["input_filter_layer"]
-                est.n_layers = self.estimators["n_layers"]
-                est.n_hidden = self.estimators["n_hidden"]
-
-            assert self.num_classes != 1
-            assert self.input_filter_layer
-        return self
 
+    def load_model(self, model_fit_params_file):
+        estimators = []
+        state_dict_list = torch.load(model_fit_params_file, map_location='cpu') #to avoid gpu ram surge
```

Comment:
I'd check this -- I think `torch.load` is returning a dict with keys like:
```
{
    n_inputs: 10,
    num_classes: 2,
    state_dicts: [LIST OF state dicts here]
}
```
If this is the case, this should actually be `easier_net_dict = torch.load...`
And the following lines should be something like
```
for sier_net_state_dict in easier_net_dict["state_dicts"]:
    sier_net_indiv = sier_net.SierNetEstimator(
        n_inputs = easier_net_dict["n_inputs"],
        num_classes = easier_net_dict["num_classes"],
         ...
    )
    sier_net_indiv.load_state_dict(sier_net_state_dict)
```

(sorry if my prev comments were wrong. hopefully we got this right now)